### PR TITLE
Add readme index for readability and fix numbering typos

### PR DESCRIPTION
**Pull request by jjelosua**
# Description

Add an index to the README since it is quite long and IMHO having an index improves readability for new comers like me.

# Test Plan

No code was touched so no test plan is required.

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:

 - [ ] I have tested these changes in the appropriate Qubes environment
 - [ ] I do not have an appropriate Qubes OS workstation set up (the reviewer will need to test these changes)
 - [x] These changes should not need testing in Qubes

If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:

 - [ ] I have updated the [AppArmor profile](https://github.com/freedomofpress/securedrop-client/blob/HEAD/files/usr.bin.securedrop-client)
 - [x] No update to the AppArmor profile is required for these changes
 - [ ] I don't know and would appreciate guidance

If these changes modify the database schema, you should include a database migration. Please check as applicable:

 - [ ] I have written a migration and upgraded a test database based on `main` and confirmed that the migration is [self-contained] and applies cleanly
 - [ ] I have written a migration but have not upgraded a test database based on `main` and would like the reviewer to do so
 - [ ] I need help writing a database migration
 - [x] No database schema changes are needed

[self-contained]: https://github.com/freedomofpress/securedrop-client#generating-and-running-database-migrations

### CODE REVIEW

**Comment by eloquence** [comment]
Index is a great idea, thanks :). I think the numbering is a matter of taste -- markdown automatically takes care of numbering when you do "1.,1." etc., and it renders correctly in the rendered view (e.g., on GitHub). Using that approach has the advantage of not having to re-number things manually :)

**Comment by jjelosua** [comment]
@eloquence glad you liked it, agree that numbering is a matter of taste, and it is clever to let Github do the auto-numbering but there were mixed uses in the readme...maybe this should be reflected in some README best practice doc to ensure homogeneity. 

I saw that a test has failed due to a timeout in CI, clearly not related to this change, maybe the load there is too high right now or the timeout too tight, I saw other recent PRs that [passed the test recently](https://app.circleci.com/pipelines/github/freedomofpress/securedrop-client/3365/workflows/48137b2b-87af-44e8-8ed7-d41efa40629c/jobs/24121)

**Review by eloquence — APPROVED** [review]
Sounds good; I'll approve but leave it open for a bit in case other folks have differing preferences re: markdown numbering.

File: README.md [hunk 1]
```diff
@@ -13,6 +13,32 @@ To learn more about architecture and our rationale behind our Qubes OS approach,
 
 **IMPORTANT:** This project is currently undergoing a pilot study and should not be used in production environments.
 
+## Index
```
Comment (cfm) [inline README.md:16]: If we do add an explicit outline, I'd think it'd be worth clarifying like so:

```suggestion
## Table of Contents
<!-- NOTE: Please keep this table of contents in sync as you add, change, and remove sections in this document! -->
```

**Review by cfm — COMMENTED** [review]
I'm agnostic on both changes here: on the renumbering for the reasons @eloquence cites; on the outline because GitHub already [generates] one from the heading tree.  But I'm certainly not opposed!  :-)

[generates]: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#headings

**Comment by gonzalo-bulnes** [comment]
> To be honest, once you know about it, the effort of maintaining it manually seems a bit redundant and likely to introduce accidental errors as the document is updated. […]

I had no idea that the table of contents was available where it is either. :open_mouth:

Not pretending to draw large inferences based on so few data points, but I wonder if it isn't worth having the Table of Contents anyway, even though it is technically redundant _once you know about that wonderful and discreet button_.

On the numbering, I personally use only `1.` (or `0.` or whatever number I want the list to start with) because it makes reordering, additions and removal slightly quicker, an diffs easier to review. That being said, we're talking about marginal gains and Markdown has the nice property of being readable in a raw form. In that latter case, having sequential numbers is an improvement without doubt.

Long story short,I have no strong feelings, but am overall in favor of the changes introduced by @jjelosua.

**Comment by gonzalo-bulnes** [comment]
Thank you @jjelosua !
